### PR TITLE
[JBIDE-26694] pre-select deployed projects by default (Deployment Page)

### PR DESCRIPTION
**Pull request by adietish**

### CODE REVIEW

**Comment by robstryker** [comment]
We originally had only deployed items showing. This was widely criticized and not appreciated by users. The reason has to do with the main workflow. 

Let me start with the basics. When you deploy or publish a project, it gets copied to your server's deployment folder. If you CHANGE a deployment name, in order to do it properly, the server must clean up the old deployment (ProjectName.war)  and then copy in with the new one (Wonka.war). 

Unfortunately, the deployment preference model does not send deltas. When you change the deployment name of a project, listeners will only be alerted that the deployment preferences for WF17ServerName has changed. They won't know which deployment had which of its properties changed. 

Since a change in deployment name has no delta, and no way for a server adapter to figure out what the previous name was, WildFly and EAP servers will not be able to clean up old deployment names (ProjectName.war) at all, if you change the deployment name after it's already been deployed.  For this reason, we do not allow the changing of deployment name for ANY published module. 

Since people cannot change the deployment name for a PUBLISHED module, then they necessarily will only be changing the deployment name for UNPUBLISHED modules. And therefore, we cannot default to only showing the deployed modules. 

If we showed the deployed modules by default, then we'd be showing the modules that the user CANNOT change or modify. The user would need to undeploy the module, re-open the editor, and then see that the deployment they wish to change the name of is not in the list. Then they'd need to figure out that they need to toggle the combo box (which users saw as obscure). 

In short, defaulting to deployed modules ends up defaulting to showing only those modules which cannot be changed, which is not user-friendly at all, and which we have tried in the past and was widely criticized as a bad workflow.

**Comment by adietish** [comment]
Showing all modules is confusing. A noob doesn't know why he gets not just the modules that are children to his server but all of them. 
Then I dont see how I CANNOT change a deployed module. I do exactly this in OpenShift. And yes, there are 2 gotchas:
* a "clean" step has to be executed before the module gets deployed to the new name
* the old folder is kept

**Comment by adietish** [comment]
closing this PR since it wont work with other adapters preventing this from being a default useable for all servers.

**Comment by adietish** [comment]
replaced by #632
